var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -163,28 +168,38 @@ def determine_b2_options(self, spec, options):
             threadingOpts.append('single')
         if not threadingOpts:
             raise RuntimeError("At least one of {singlethreaded, multithreaded} must be enabled")
-        
+
         options.extend([
             'toolset=%s' % self.determine_toolset(spec),
             'link=%s' % ','.join(linkTypes),
             '--layout=tagged'])
-        
+
         return threadingOpts
 
     def install(self, spec, prefix):
+        # On Darwin, Boost expects the Darwin libtool. However, one of the
+        # dependencies may have pulled in Spack's GNU libtool, and these two are
+        # not compatible. We thus create a symlink to Darwin's libtool and add
+        # it at the beginning of PATH.
+        if sys.platform == 'darwin':
```

Comment:
why exactly are you against of global usage of Darwin's libtool, if I may ask? So far I have installed quite a number of things including `metis`, `parmetis`, `mumps`, `trilinos`, `petsc` and others without GNU libtool. Is there any package which fails for you with Darwin libtool?

I think the changes you propose are good to make things work, but it would still be better to also provide Mac-specific `package.yaml` which uses as much as possible from XCode, including `libtool`, `zlib`, `blas/lapack`.